.github/workflows/build_and_test.yml

```diff
@@ -1,28 +1,41 @@
 name: Build and test
 on:
   push:
-    branches: [ "master", "dev" ]
+    branches: ["master", "dev"]
   pull_request:
-    branches: [ "dev" ]
+    branches: ["dev"]
 
 env:
   BUILD_DIR: Dist
 
 jobs:
   pre-commit:
     runs-on: ubuntu-latest
+    container: seladb/alpine317
```

Comment:
Why is it needed to run `pre-commit` in a container? 🤔

Reply:
we need cmake, clang, clang-extra-tools

Reply:
ah ok, that's because we enable `clang-tidy`